Add Aqara Wall Outlet H2 EU

andreibsk opened the pull request:
## Proposed change
<!--
  Explain your proposed change below.
-->
This PR should add new configuration options and remove unused entities for Aqara Wall Outlet H2 EU (Aqara lumi.plug.aeu001) using the quirks v2 api.


## Additional information
<!--
  Please include any additional information that is important to this PR.
  For example, if this PR is a potentially breaking change, mention that here.
  If this PR requires other PRs to be merged in HA Core or other projects, mention that.
  Lastly, if this PR fixes a specific issue, please include "Fixes #xxxx".
-->
This is my first time writing a quirk and first time coding in Python, so please advise where improvements are possible.

Changes:
* removed temperature as it wasn't working as expected
* removed unknown/unused switch
* added Button lock
* added Charging Protection
* added option to configure LED indicator
* added Overload Protection
* added Power on behaviour

All the new configuration options have been tested on a device I own.

Related: #3265 #3187 

## Checklist
<!--
  Put an 'x' in all boxes that apply.
  Note: You do not need to tick all boxes before creating a PR.
-->

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [x] Tests have been added to verify that the new code works

andreibsk commented:
Here's a screenshot of how it looks in my setup:


![image](https://github.com/user-attachments/assets/c4ab228e-79f9-446d-a1ad-0224f574cd42)

andreibsk commented:
I noticed that for 1 out of my 5 devices the 'Summation delivered' reading was fluctuating up and down causing warnings and wrong data in the Energy dashboard.
This was because ElectricalMeasurementCluster received values by
`Received command 0x0A (TSN 31): Report_Attributes(attribute_reports=[Attribute(attrid=0x0000, value=TypeValue(type=uint48_t, value=23469))])`
and additionally from XiaomiCluster -> XIAOMI_AQARA_ATTRIBUTE_E1 -> CONSUMPTION 
https://github.com/zigpy/zha-device-handlers/blob/64c3348c3b42897b6118b4e83c2d5eb32b360aeb/zhaquirks/xiaomi/__init__.py#L303

The former one seems to lag behind in updating the summ and still sends the outdated value. The latter one sends values that are up to date. This leads the aggregated value switching between the outdated and the actual value:
<img width="433" alt="image" src="https://github.com/user-attachments/assets/98fec1f5-79ce-4ac9-be66-d3782a97c8b6" />

To avoid this, I've added the `PlugAEU001MeteringCluster` in the latest commit. Please let me know if this an appropriate solution or if there are better approaches.
https://github.com/zigpy/zha-device-handlers/pull/3653/commits/a8e6e3c3f56a1285a629737a2464e744219f6880

TheJulianJES commented on zhaquirks/xiaomi/aqara/plug_eu.py:
```diff
@@ -346,3 +354,102 @@ class PlugMAEU01Alt3(PlugMAEU01):
+        cluster_id=PlugAEU001Cluster.cluster_id,
+        force_inverted=True,
+        translation_key="button_lock",
+        fallback_name="Button Lock",
```
Per HA spec, only the first letter should be capitalized, all other words should be lowercase (except for abbreviations). So, this should be "Buton lock", although we may want to use `child_lock` for the `translation_key` (and `fallback_name` if that is the same, as that's what we use for other devices).

Please also adjust the names of the other entities. (E.g. "Power on behavior", "LED indicator", ...) Thanks!

andreibsk replied:
Changed to `child_lock` and updated the other names as well. 👍

TheJulianJES commented on zhaquirks/xiaomi/aqara/plug_eu.py:
```diff
@@ -346,3 +354,102 @@ class PlugMAEU01Alt3(PlugMAEU01):
+    }
+
+
+class PlugAEU001MeteringCluster(MeteringCluster):
+    """Custom cluster for Aqara lumi plug AEU001."""
+
+    def _update_attribute(self, attrid, value):
+        if attrid == self.CURRENT_SUMM_DELIVERED_ID:
+            current_value = self._attr_cache.get(attrid, 0)
+            if value < current_value:
+                _LOGGER.debug(
+                    "Ignoring attribute update for %s: new value %s is less than current value %s",
+                    attrid,
+                    value,
+                    current_value,
+                )
+                return
+        super()._update_attribute(attrid, value)
```
Hmm, I'll have to check back on this. If the custom Xiaomi reports are more up-to-date/accurate and are also always sent, we could just ignore the ZCL reports coming in by overriding `handle_cluster_general_request` like this: https://github.com/zigpy/zha-device-handlers/blob/57bcec8cf963499b9fca76484619ec57e4a7c088/zhaquirks/xiaomi/__init__.py#L568-L579

That would ignore all attribute reports from the EM cluster though, since we essentially remove all of this: https://github.com/zigpy/zigpy/blob/07e0451ce5317185e04b38e43aeb9fb168c62f28/zigpy/zcl/__init__.py#L464-L530

We could probably also block the calls to set up attribute reporting for that attribute, but we should still block incoming attribute reports, as some devices may have been connected before.

I'll need to come back to this. Hopefully soon.

andreibsk replied:
Let me know if/how I could help in this regard.

andreibsk replied:
Hey @TheJulianJES, how do you think we could move forward on this?

TheJulianJES replied:
Can you just paste that empty `handle_cluster_general_request` method (with just the docstring) in your quirk to test if you still have the fluctuations (but proper reporting) then? So, removing your `_update_attribute` then.

andreibsk replied:
Removing `_update_attribute` and replacing it with an empty `handle_cluster_general_request` leads to a loss of reporting on multiple other entities (see pic) defined on the other cluster `PlugAEU001Cluster(XiaomiAqaraE1Cluster)`. 
Sorry, it's been a while for me, and I need to refamiliarize myself with how things work. Is this because of the LocalDataCluster (from MeteringCluster) sending values to another cluster?
How can we handle this?

<img width="244" alt="image" src="https://github.com/user-attachments/assets/7c546ee0-30c2-492f-a1ab-c86f31ed4918" />

TheJulianJES commented on zhaquirks/xiaomi/aqara/plug_eu.py:
```diff
@@ -346,3 +354,102 @@ class PlugMAEU01Alt3(PlugMAEU01):
+        0x0206: ("charging_limit", types.Single, True),
+        0x020B: ("overload_protection", types.Single, True),
+        0x0517: ("power_on_behavior", AqaraPowerOutageMemoryEnum, True),
+    }
```
Please use new zigpy style `AttributeDefs`. Examples: https://github.com/zigpy/zha-device-handlers/blob/c4e2a1cf48cda79dcc505913bbfd3296e6c79154/zhaquirks/thirdreality/plug.py#L11-L23 and https://github.com/zigpy/zha-device-handlers/blob/c4e2a1cf48cda79dcc505913bbfd3296e6c79154/zhaquirks/ikea/vindstyrka.py#L18-L36

andreibsk replied:
Updated.

ChristophCaina commented:
can this be reviewed?
It would be great, if this device would be fully supported after nearly 9 months :)

ChristophCaina commented:
I've implemented this as custom quirk now and noticed the following:

1. it will be detected as expected (Wall Outlet H2 EU)
2. "Watt" usage will be shown correctly
3. "Voltage" remains at 0.0
4. "Summation delivered" (kWh) requires quite a while to be updated, but I need to monitor, if it will be updated more frequently, compared to earlier
5. Temperature has been moved to diagnostic sensors, but is providing data now.
6. other entities, such as charging-protection, child-protection, led and overload-protection are also available now.

Seems only the Voltage Sensor, which is not yet working.

Update: 
Also the kWh Sensor did not Change within a few hours, while the dishwasher was running... :-(

ChristophCaina commented:
[update]
Using the Aqara Hub, I can confirm that the Plug is working & reporting as expected.

Unfortunately, the Screenshot from @andreibsk does not reveal, which Fimrware-Version is running on his Plug.
Mine has 0.0.0_0030 [0x1E in ZHA].
There's no newer Firmware available.

Since I am getting the expected reports within the Aqara Hub, but not via this ZHA quirk, could indicate, that something within the Firmware version might have changed - and therefore, this quirk isn't working as expected (?)

ChristophCaina commented:
looking into my logs,
I am getting the following information:

```
2025-10-08 13:32:43.051 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 37, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.178266525268555, '0xff01-150': 2361.10009765625, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.95000076293945, '0xff01-154': 0}]

2025-10-08 13:40:41.531 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 36, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.178865909576416, '0xff01-150': 2377.89990234375, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.77000045776367, '0xff01-154': 0}]

2025-10-08 13:46:12.522 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 34, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.179165840148926, '0xff01-150': 2364.60009765625, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.77000045776367, '0xff01-154': 0}]

2025-10-08 13:53:25.675 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 33, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.1795654296875, '0xff01-150': 2371.60009765625, '0xff01-152': 3.75, '0xff01-151': 55.59000015258789, '0xff01-154': 0}]

2025-10-08 14:01:01.759 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 33, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.179965496063232, '0xff01-150': 2365.300048828125, '0xff01-152': 0.0, '0xff01-151': 0.0, '0xff01-154': 0}]



2025-10-09 10:04:09.007 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 30, 'X-attrib-5': 84, '0xff01-9': 5376, 'path': 36486, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 0.0, '0xff01-150': 2343.60009765625, '0xff01-152': 0.07999999821186066, '0xff01-151': 72.16999816894531, '0xff01-154': 0}]

2025-10-09 10:09:35.811 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 30, 'X-attrib-5': 84, '0xff01-9': 5376, 'path': 36486, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 0.0, '0xff01-150': 2345.0, '0xff01-152': 0.0, '0xff01-151': 0.0, '0xff01-154': 0}]

```

it seems, that voltage will be reported in this cluster:
'0xff01-150': 2343.60009765625
'0xff01-150': 2345.0

this seems to be the Watt, which will match the graph in HomeAssistant (will be read correctly)
'0xff01-152': 3.75,

kWh seems to be reported within this attribute:
'0xff01-149':

Here's an image I've rendered based on a larger log-output.
The gap in the chart is the time-frame, when I removed the plug from ZHA to test it with the aqara app.

After adding it again, the kWh got resetted.
<img width="1400" height="1000" alt="image" src="https://github.com/user-attachments/assets/374a2b3b-a49b-4294-87a8-ce5abf5c426d" />

andreibsk committed:
Ignore summation delivered if lower value received

andreibsk committed:
Add Aqara Wall Outlet H2 EU

andreibsk committed:
Update entity names

andreibsk committed:
Use AttributeDefs

andreibsk committed:
Add device temperature

andreibsk committed:
Add test for AEU001 Metering

andreibsk commented:
> looking into my logs, I am getting the following information:
> 
> ```
> 2025-10-08 13:32:43.051 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 37, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.178266525268555, '0xff01-150': 2361.10009765625, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.95000076293945, '0xff01-154': 0}]
> 
> 2025-10-08 13:40:41.531 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 36, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.178865909576416, '0xff01-150': 2377.89990234375, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.77000045776367, '0xff01-154': 0}]
> 
> 2025-10-08 13:46:12.522 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 34, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.179165840148926, '0xff01-150': 2364.60009765625, '0xff01-152': 3.6700000762939453, '0xff01-151': 55.77000045776367, '0xff01-154': 0}]
> 
> 2025-10-08 13:53:25.675 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 33, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.1795654296875, '0xff01-150': 2371.60009765625, '0xff01-152': 3.75, '0xff01-151': 55.59000015258789, '0xff01-154': 0}]
> 
> 2025-10-08 14:01:01.759 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 33, 'X-attrib-5': 81, '0xff01-9': 5376, 'path': 0, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 7.179965496063232, '0xff01-150': 2365.300048828125, '0xff01-152': 0.0, '0xff01-151': 0.0, '0xff01-154': 0}]
> 
> 
> 
> 2025-10-09 10:04:09.007 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 30, 'X-attrib-5': 84, '0xff01-9': 5376, 'path': 36486, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 0.0, '0xff01-150': 2343.60009765625, '0xff01-152': 0.07999999821186066, '0xff01-151': 72.16999816894531, '0xff01-154': 0}]
> 
> 2025-10-09 10:09:35.811 DEBUG (MainThread) [zhaquirks.xiaomi] 54:ef:44:10:00:c9:cc:85 - Xiaomi attribute report. attribute_id: [247] value: [{'temperature': 30, 'X-attrib-5': 84, '0xff01-9': 5376, 'path': 36486, '0xff01-12': 10, '0xff01-13': 30, '0xff01-17': 1, '0xff01-100': <Bool.true: 1>, '0xff01-101': <Bool.false: 0>, '0xff01-149': 0.0, '0xff01-150': 2345.0, '0xff01-152': 0.0, '0xff01-151': 0.0, '0xff01-154': 0}]
> ```
> 
> it seems, that voltage will be reported in this cluster: '0xff01-150': 2343.60009765625 '0xff01-150': 2345.0
> 
> this seems to be the Watt, which will match the graph in HomeAssistant (will be read correctly) '0xff01-152': 3.75,
> 
> kWh seems to be reported within this attribute: '0xff01-149':
> 
> Here's an image I've rendered based on a larger log-output. The gap in the chart is the time-frame, when I removed the plug from ZHA to test it with the aqara app.
> 
> After adding it again, the kWh got resetted. <img alt="image" width="1400" height="1000" src="https://private-user-images.githubusercontent.com/26391061/499322171-374a2b3b-a49b-4294-87a8-ce5abf5c426d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjAxNjkxNzUsIm5iZiI6MTc2MDE2ODg3NSwicGF0aCI6Ii8yNjM5MTA2MS80OTkzMjIxNzEtMzc0YTJiM2ItYTQ5Yi00Mjk0LTg3YTgtY2U1YWJmNWM0MjZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDExVDA3NDc1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjZDk5MDEzNzY0NDJiOGMzMDlmMTkwOWVmYWRlNTNmMzZlMjkzMjkwOWZkZTJjODdlNmFjMGY4NDZlY2VhMWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8c-HnvxA3E9LYYZ3YfyXc0XHa-YcaX14nD0q1ms5w2g">

Hello @ChristophCaina 
Regarding your testing by adding this as a custom quirk, I'm wondering if perhaps you're missing the change from this PR in [xiaomi/__init__.py](https://github.com/zigpy/zha-device-handlers/pull/3653/files#diff-e4e4ec70dd92e877068f1842b8a76afcde44d9365bce97a890fb90fab45f2bd5) at line 407. That piece handle additional updates for `update({149: CONSUMPTION, 150: VOLTAGE, 152: POWER})`. If that's the case, please try it out after adding this change as well, and let me know how it goes.

Regarding the firmware versions, I have 6 H2 EU outlets, and all of them are on firmware `0x0000001e` - so same as you.

andreibsk commented:
@TheJulianJES Do you think you would have some time in the near future to revisit this PR and see what we can do to get it integrated?
I've also added a small test for that custom metering cluster.
Let me know what else needs to be done.

ChristophCaina commented:
Hi, 
Thanks für your Response.
Since this is the first time I am using a custom quirk, it might be that I made a mistake ...

Init File and quirk file should be in the same folder?

andreibsk commented:
> Hi, Thanks für your Response. Since this is the first time I am using a custom quirk, it might be that I made a mistake ...
> 
> Init File and quirk file should be in the same folder?

So I'm assuming that you created a custom quirk file, pasted the quirk from the PR, and configured zha to load quirks from that folder.
Well this quirk uses this `PlugAEU001Cluster` class which references `XiaomiAqaraE1Cluster` class from the internal library files, which references `XiaomiCluster` class, also internal. In this last class is the one-line change mentioned above.

One way I did tests in the past which included this change, was to directly edit this file in my HA container instance and add the `"lumi.plug.aeu001",` line there + restart. The file was situated in `/usr/local/lib/python3.13/site-packages/zhaquirks/xiaomi/__init__.py`. However, the approach of editing such files might not be as advisable, but it's quick.

A different approach would be to make a copy of the two classes into your own quirk file and reference those. Make sure other constant values and dependencies and imports are referenced appropriately. eg. `PlugAEU001Cluster -> XiaomiAqaraE1ClusterModified -> XiaomiClusterModified`

ChristophCaina commented:
thanks, I'll have a look into your suggestions :)

ChristophCaina commented:
sorry for the delayed response on this.

finally, I got this working as a custom quirk,
I am getting the power information, the kWh measurement ...

It is also showing the temperature and several controls and setting options.
In addition to that, I got a new Sensor "immediate need" in Watt.

What's still not working seems to be the voltage measurement.
At least, this is showing 0.0 V... 

<img width="325" height="518" alt="image" src="https://github.com/user-attachments/assets/8e0c762a-6c01-472b-8738-5b3c1ae68670" />
<img width="324" height="835" alt="image" src="https://github.com/user-attachments/assets/f3536d7d-261e-438e-b8f8-092b752d55b2" />

andreibsk commented:
> sorry for the delayed response on this.
> 
> finally, I got this working as a custom quirk, I am getting the power information, the kWh measurement ...
> 
> It is also showing the temperature and several controls and setting options. In addition to that, I got a new Sensor "immediate need" in Watt.
> 
> What's still not working seems to be the voltage measurement. At least, this is showing 0.0 V...
> 
> <img alt="image" width="325" height="518" src="https://private-user-images.githubusercontent.com/26391061/501949064-8e0c762a-6c01-472b-8738-5b3c1ae68670.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA4ODYzNzksIm5iZiI6MTc2MDg4NjA3OSwicGF0aCI6Ii8yNjM5MTA2MS81MDE5NDkwNjQtOGUwYzc2MmEtNmMwMS00NzJiLTg3MzgtNWIzYzFhZTY4NjcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE5VDE1MDExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkNTgxOTJhZmQ4OTQ4ZGM2NzNiMDMwOTZmODNmZWEzNDUxMjFlNzM0MDc5ZDM2NzdlZmE3ZjQxZTAwYzZjYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Psr_XSpV6uGgcTY2fA29aAcbC_yPsnG_MiBgFobO-NI"> <img alt="image" width="324" height="835" src="https://private-user-images.githubusercontent.com/26391061/501949227-f3536d7d-261e-438e-b8f8-092b752d55b2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA4ODYzNzksIm5iZiI6MTc2MDg4NjA3OSwicGF0aCI6Ii8yNjM5MTA2MS81MDE5NDkyMjctZjM1MzZkN2QtMjYxZS00MzhlLWI4ZjgtMDkyYjc1MmQ1NWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE5VDE1MDExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjNDQ5YWI1NWMzNzFhYTM1YmE1N2UzMDAxZjdmYjMxNWVkYjU4OGFlMzZjMDYyYzhiZTg1ZmRjYWQ1ZjliOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rFShz0nVsCBK2gI1Tjs35U7bnT3LJ96WNsnbPt1Gzow">

@ChristophCaina 
For me the "immediate need" entity disappears after a while, or perhaps I removed it manually, can't remember. Not sure why it appeared in the first place.
As to the voltage entity, please wait a few minutes for it to receive some values and appear.

I've created a gist with how I added the quirk in my quirks directory with the modified Xiaomi clusters.
See: https://gist.github.com/andreibsk/fa9d68a417430bb4d00ff592345866de#file-aqara_wall_outlet_h2_eu-py

Let me know how if you see the entities as expected.

tasomaniac commented:
What does charging limit and protection mean?

andreibsk commented:
> What does charging limit and protection mean?

If I remember correctly, it's for switching off the outlet when your phone is done charging, but still sips a bit of power being connected to the charger.

ChristophCaina commented:
> @ChristophCaina For me the "immediate need" entity disappears after a while, or perhaps I removed it manually, can't remember. Not sure why it appeared in the first place. As to the voltage entity, please wait a few minutes for it to receive some values and appear.
> 
> I've created a gist with how I added the quirk in my quirks directory with the modified Xiaomi clusters. See: https://gist.github.com/andreibsk/fa9d68a417430bb4d00ff592345866de#file-aqara_wall_outlet_h2_eu-py
> 
> Let me know how if you see the entities as expected.

Hi,
Sensors appear as they did appear with my modification.
But after short time, also voltage did show values, so I probably forgot some of the related imports ... thanks a lot :)

ChristophCaina commented:
will this ever be merged?

ChristophCaina commented:
nearly April and still pending the review to be merged.
I understand now, why people prefer z2mqtt over zha... :-(
